nemo/collections/asr/parts/submodules/rnnt_beam_decoding.py

```diff
@@ -1319,21 +1365,49 @@ def prefix_search(
                         self.joint.joint(enc_out, hyp_i.dec_out[-1]) / self.softmax_temperature, dim=-1,
                     )
                     logp = logp[0, 0, 0, :]
-
                     curr_score = hyp_i.score + float(logp[hyp_j.y_sequence[pref_id]])
+                    # Setup ngram LM:
+                    if self.ngram_lm:
+                        lm_score, next_state = self.compute_ngram_score(
+                            hyp_i.ngram_lm_state, int(hyp_j.y_sequence[pref_id])
+                        )
+                        curr_score += self.ngram_lm_alpha * lm_score
 
                     for k in range(pref_id, (curr_id - 1)):
                         logp = torch.log_softmax(
                             self.joint.joint(enc_out, hyp_j.dec_out[k]) / self.softmax_temperature, dim=-1,
                         )
                         logp = logp[0, 0, 0, :]
-
                         curr_score += float(logp[hyp_j.y_sequence[k + 1]])
+                        # Setup ngram LM:
+                        if self.ngram_lm:
+                            lm_score, next_state = self.compute_ngram_score(next_state, int(hyp_j.y_sequence[k + 1]))
+                            curr_score += self.ngram_lm_alpha * lm_score
 
                     hyp_j.score = np.logaddexp(hyp_j.score, curr_score)
 
         return hypotheses
 
+    def compute_ngram_score(self, current_lm_state: "kenlm.State", label: int) -> Tuple[float, "kenlm.State"]:
+        """
+        Score computation for kenlm ngram language model.
+        """
+
+        next_state = kenlm.State()
+        lm_score = self.ngram_lm.BaseScore(current_lm_state, chr(label + self.token_offset), next_state)
```

Comment:
Token offset should only be applied for subword lms not char

Reply:
Yes, it was the dug. I fixed it.